bundles/org.pitest.pitclipse.ui/src/org/pitest/pitclipse/ui/view/PitView.java

```diff
@@ -16,28 +16,73 @@
 
 package org.pitest.pitclipse.ui.view;
 
+import org.eclipse.core.resources.IFile;
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.action.IToolBarManager;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.SWTError;
 import org.eclipse.swt.browser.Browser;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.MessageBox;
+import org.eclipse.ui.IActionBars;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.part.FileEditorInput;
 import org.eclipse.ui.part.ViewPart;
 
 import java.io.File;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
  * A view displaying the HTML report generated by PIT.
  */
 public class PitView extends ViewPart implements SummaryView {
     private Browser browser = null;
     private PitUiUpdatePublisher publisher = null;
+    private String homeUrlString = null;
+    private Action homeAction;
+    private Action backButton;
+    private Action forwardButton;
 
     @Override
     public synchronized void createPartControl(Composite parent) {
         try {
             browser = new Browser(parent, SWT.NONE);
             publisher = new PitUiUpdatePublisher(browser);
             browser.addProgressListener(publisher);
+            // create back button
+            backButton = new Action("<") {
+                @Override
+                public void run() {
+                    if (isBrowserSet()) {
+                        browser.back();
+                    }
+                }
+            };
+            // create home button for navigation
+            homeAction = new Action("Home") {
+                @Override
+                public void run() {
+                    if (isBrowserSet()) {
+                        browser.setUrl(homeUrlString);
+                    }
+                }
+            };
+            // create forward button
+            forwardButton = new Action(">") {
+                @Override
+                public void run() {
+                    if (isBrowserSet()) {
+                        browser.forward();
+                    }
+                }
+            };
+            IActionBars actionBars = getViewSite().getActionBars();
+            IToolBarManager toolBar = actionBars.getToolBarManager();
+            toolBar.add(backButton);
```

Comment:
If I understand correctly (I haven't tried that locally), the buttons are added to the Eclipse toolbar, not the the PIT view?

Reply:
It is added to the view itself:
![grafik](https://user-images.githubusercontent.com/50097340/125592556-12436e94-042f-4211-ad7a-5f2d0cfb01d7.png)

Reply:
OK!